test/cypress/integration/all-service-transactions/all-service-transactions.cy.test.js

```diff
@@ -0,0 +1,115 @@
+  const gatewayAccountExternalId2 = 'a-valid-external-id-2'
+  const transactionsUrl = `/all-service-transactions`
+  const defaultAmount = 1000
+
+  const defaultTransactionEvents = [{
+    amount: defaultAmount,
+    state: {
+      finished: false,
+      status: 'created'
+    },
+    resource_type: 'PAYMENT',
+    event_type: 'PAYMENT_CREATED',
+    timestamp: '2019-09-18T10:06:17.152Z',
+    data: {}
+  }]
+
+  function generateTransactions (length) {
+    const transactions = []
+    for (let i = 0; i < length; i++) {
+      transactions.push({
+        reference: 'transaction' + i,
+        amount: defaultAmount,
+        type: 'payment',
+        transaction_id: 'transaction-id-' + i,
+        gateway_account_id: String(gatewayAccountId1),
+        events: defaultTransactionEvents
+      })
+    }
+    return transactions
+  }
+
+  function transactionSearchResultOpts (transactionLength, displaySize, page, filters, links) {
+    return {
+      gatewayAccountIds: [ gatewayAccountId1, gatewayAccountId2 ],
+      transactionLength: transactionLength || 50,
+      displaySize: displaySize || 5,
+      page: page || 1,
+      transactionCount: 3,
+      transactions: generateTransactions(2),
+      filters: filters,
+      links: links || {}
+    }
+  }
+
+  describe('All Service Transactions', () => {
+    beforeEach(() => {
+      Cypress.Cookies.preserveOnce('session', 'gateway_account')
+    })
+
+    it('should display All Service Transactions > list page', () => {
+      cy.setEncryptedCookies(userExternalId, gatewayAccountId1)
+      const opts = transactionSearchResultOpts(30, 5, 1, {},
+        {
+          self: { href: '/v1/transactions?&page=&display_size=5&state=' },
+          next_page: { href: '/v1/transactions?&page=3&display_size=5&state=' }
+        })
+
+      cy.task('setupStubs', [
+        userStubs.getUserSuccessWithMultipleServices({ userExternalId, gatewayAccountId1, gatewayAccountId2, gatewayAccountExternalId1, gatewayAccountExternalId2 }),
+        gatewayAccountStubs.getGatewayAccountsSuccessForMultipleAccounts({ gatewayAccountIds: [gatewayAccountId1, gatewayAccountId2], gatewayAccountId1, gatewayAccountId2, gatewayAccountExternalId1, gatewayAccountExternalId2, type: 'live', paymentProvider: 'stripe' }),
+        transactionStubs.getLedgerTransactionsSuccess(opts),
+        gatewayAccountStubs.getCardTypesSuccess()
+      ])
+
+      cy.visit(transactionsUrl + '?pageSize=5&page=')
+      cy.title().should('eq', `Transactions for all live services`)
+
+      cy.get('form.paginationForm.page-Previous').should('exist').within(() => {
+        cy.get('input[name="page"]').should('have.value', '')
+      })
+      cy.get('button.pagination.Previous').should('exist')
+      cy.get('button.pagination.Previous').should('be.disabled')
+
+      cy.get('form.paginationForm.page-Next').should('exist').within(() => {
+        cy.get('input[name="page"]').should('have.value', '2')
+      })
```

Comment:
I think we have another Cypress test that checks the pagination, and the all services transactions shares the same template. Reckon we could remove these pagination assertions so this test is a bit more focussed?

Reply:
Good point - we forgot to do that.